### DISCUSSION TITLE
NoPresentationContextException: No Presentation Context for Abstract Syntax: 1.2.840.10008.5.1.4.1.1.1.1 - Digital X-Ray Image Storage - For Presentation with Transfer Syntax: 1.2.840.10008.5.1.4.1.1.1.1 - Digital X-Ray Image Storage - For Presentation negotiated

### DISCUSSION THREAD

**Discussion by advanceMaster** [dcm4che/dcm4che#1012] [Q&A]
version:5.22.6
When I set up a DcmQRSCP for local testing, if I use getSCU to test getSCP, everything works. But when I test getSCU with a third-party tool called getSCU(e.g. Stante PACS Server, MicroDicom Viewer), I get an exception:
Unable to perform sub-operation on association to SANTESRV1
org.dcm4che3.net.NoPresentationContextException: No Presentation Context for Abstract Syntax: 1.2.840.10008.5.1.4.1.1.1.1 - Digital X - Ray Image Storage - For the Presentation with Transfer Syntax: 1.2.840.10008.5.1.4.1.1.1.1 - Digital X - Ray Image Storage - For the Presentation negotiated
at org.dcm4che3.net.Association.pcFor(Association.java:902)
at org.dcm4che3.net.Association.cstore(Association.java:925)
at org.dcm4che3.net.service.BasicRetrieveTask.cstore(BasicRetrieveTask.java:258)
at org.dcm4che3.net.service.BasicRetrieveTask.run(BasicRetrieveTask.java:185)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
at java.lang.Thread.run(Thread.java:745)
i/o exception: java.io.EOFException in State: Sta6 - Association established and ready for data transfer
MoveSCP is all right

**Comment by gunterze** [comment]
dcm4che's [getscu utility](https://github.com/dcm4che/dcm4che/blob/master/dcm4che-tool/dcm4che-tool-getscu/README.md) offers a Presentation Context for receive of `1.2.840.10008.5.1.4.1.1.1.1 - Digital X-Ray Image Storage - For Presentation` [by default](https://github.com/dcm4che/dcm4che/blob/master/dcm4che-assembly/src/etc/getscu/store-tcs.properties#L12), which may not be the case by DICOM getscu utilities of other DICOM toolkits.

**Reply by advanceMaster** [reply]
This is the log of the other SCU 
# Sending C-GET-RQ to AE Title "XTPACS"...
    - C-GET-RQ was sent
        - Message Contents:
          Group	 Elem	 VR	 Size	 Data
          0000	 0000	 UL	 4	 76
          0000	 0002	 UI	 28	 1.2.840.10008.5.1.4.1.2.2.3
          0000	 0100	 US	 2	 10H
          0000	 0110	 US	 2	 4H
          0000	 0700	 US	 2	 0H
          0000	 0800	 US	 2	 102H

        - Message Contents:
          Group	 Elem	 VR	 Size	 Data
          0020	 000D	 UI	 56	 1.2.840.10008.5.1.4.31.681571745.2603640.1425283467.955

# Waiting for C-GET-RSP...
    - C-GET-RSP was received from AE Title "XTPACS"
        - Message Contents:
          Group	 Elem	 VR	 Size	 Data
          0000	 0000	 UL	 4	 116
          0000	 0002	 UI	 28	 1.2.840.10008.5.1.4.1.2.2.3
          0000	 0100	 US	 2	 8010H
          0000	 0120	 US	 2	 4H
          0000	 0800	 US	 2	 101H
          0000	 0900	 US	 2	 FF00H
          0000	 1020	 US	 2	 2H
          0000	 1021	 US	 2	 0H
          0000	 1022	 US	 2	 0H
          0000	 1023	 US	 2	 0H
#  0 completed file(s)
# Service time-out
# Connection is closed (Service Duration: 63 sec)

This is my SCP log

XTPACS<-SANTESRV1(4) << 4:C-GET-RSP[pcid=1, remaining=2, completed=0, failed=0, warning=0, status=ff00H
  cuid=1.2.840.10008.5.1.4.1.2.2.3 - Study Root Query/Retrieve Information Model - GET
  tsuid=1.2.840.10008.1.2.1 - Explicit VR Little Endian]
Successfully loaded OpenCV native library.
 XTPACS<-SANTESRV1(4): Unable to perform sub-operation on association to SANTESRV1

org.dcm4che3.net.NoRoleSelectionException: No Role Selection for SOP Class 1.2.840.10008.5.1.4.1.1.1.1 - Digital X-Ray Image Storage - For Presentation as SCU negotiated
	at org.dcm4che3.net.Association.checkIsSCU(Association.java:265)
	at org.dcm4che3.net.Association.cstore(Association.java:926)
	at org.dcm4che3.net.service.BasicRetrieveTask.cstore(BasicRetrieveTask.java:258)
	at org.dcm4che3.net.service.BasicRetrieveTask.run(BasicRetrieveTask.java:185)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

XTPACS<-SANTESRV1(4) << 4:C-GET-RSP[pcid=1, completed=0, failed=2, warning=0, status=a702H
  cuid=1.2.840.10008.5.1.4.1.2.2.3 - Study Root Query/Retrieve Information Model - GET
  tsuid=1.2.840.10008.1.2.1 - Explicit VR Little Endian]
XTPACS<-SANTESRV1(4): **i/o exception: java.io.EOFException in State: Sta6 - Association established and ready for data transfer**
XTPACS<-SANTESRV1(4): close Socket[addr=/127.0.0.1,port=53429,localport=106]

All types, not just DX, have current problems

**Reply by advanceMaster** [reply]
No Role Selection for SOP Class 1.2.840.10008.5.1.4.1.1.4 - MR Image Storage as SCU negotiated
No Role Selection for SOP Class 1.2.840.10008.5.1.4.1.1.2 - CT Image Storage as SCU negotiated

**Reply by gunterze** [reply]
The C-GET SCU has to offer support of the Storage SOP Classes as SCP in the A-ASSOCIATE RQ, and the C-GET SCP has to accept that Role Selections for the Storage SOP Classes in returned  A-ASSOCIATE SC. s. [DICOM PS.3.7 D.3.3.4. SCP/SCU Role Selection Negotiation](http://dicom.nema.org/medical/dicom/current/output/html/part07.html#sect_D.3.3.4)
